rholang/src/main/scala/coop/rchain/rholang/interpreter/Interpreter.scala

```diff
@@ -60,30 +68,55 @@ object Interpreter {
         sortedPar <- Sortable[Par].sortMatch(par)
       } yield sortedPar.term
 
-    def execute(runtime: Runtime[F], reader: Reader): F[Runtime[F]] =
-      for {
-        par    <- buildNormalizedTerm(reader)
-        errors <- evaluate(runtime, par).map(_.errors)
-        result <- if (errors.isEmpty) F.pure(runtime)
-                 else F.raiseError(new RuntimeException(mkErrorMsg(errors)))
-      } yield result
+    def evaluatePar(runtime: Runtime[F], par: Par): F[EvaluateResult] = {
+      val initialPhlo = Cost.Max
+      evaluatePar(runtime, par, initialPhlo)
+    }
+
+    def evaluatePar(runtime: Runtime[F], par: Par, initialPhlo: Cost): F[EvaluateResult] = {
+      val term = PrettyPrinter().buildString(par)
+      evaluate(runtime, term, initialPhlo)
```

Comment:
we really shouldn't be pretty-printing an already parsed par and parsing it again in evaluate. Not until we have property-based tests checking a `.rho -> Par -> .rho -> Par` roundtrip to arrive at the same Par twice. I'm pretty sure the PrettyPrinter is not robust enough for such usage, even if we take aside the inefficiency this causes.

Reply:
How would you approach .rho file generation?

Reply:
Tymm is working on it

Reply:
while he is though, we need to change this - if not for just the inefficiency this introduces

Reply:
This function is used only in tests and both repls so efficiency shouldn't be top concern here

Reply:
the real issue is correctness though. Also, this is just a matter of reversing the delegation order. So I don't see why are we even discussing this.

Reply:
It's not a matter of reversing the delegation order because `injAttempt` expects `term: String`

Reply:
That's exactly my point. Make it expect a Par (and make it private?), and introduce a public wrapper that does the parsing to retain the callsites.

Reply:
I won't be able to calculate the parsing cost from an already parsed term

Reply:
Then (1) pass it from the outside (could be dummy if in test code), or - better yet - (2) replace the usages of evaluatePar with evaluate. All call sites but `CostAccountingPropertyTest` have the unparsed rho as well not so far above them. So please do (1) for `CostAccountingPropertyTest` and (2) for the other places.